### PR TITLE
SAK-41295 Course grade miscalculates if there's an Extra Credit cate…

### PR DESCRIPTION
**Pull request by jesusmmp**
…gory and 1 or more other categories are empty

### CODE REVIEW

**Comment by jesusmmp** [comment]
I've updated this PR changing double with BigDecimal. Please check it.
Thanks!

File: edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java [hunk 1]
```diff
@@ -217,20 +217,23 @@ public void setCalculatedPointsEarned(final double literalTotalPointsEarned) {
...
 		final BigDecimal bdTotalPointsPossible = new BigDecimal(totalPointsPossible);
 		final BigDecimal bdTotalPointsEarned = new BigDecimal(totalPointsEarned);
 
 		if (totalPointsPossible <= 0.0) {
-			percentageEarned = null;
+			this.autoCalculatedGrade = null;
 		} else {
-			percentageEarned = Double.valueOf(bdTotalPointsEarned.divide(bdTotalPointsPossible, GradebookService.MATH_CONTEXT)
-					.multiply(new BigDecimal("100")).doubleValue());
+			percentageEarned =  bdTotalPointsEarned.divide(bdTotalPointsPossible, GradebookService.MATH_CONTEXT)
```
Comment (jesusmmp) [inline edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java]: Would I use MathContext.DECIMAL128? like in this one
https://github.com/sakaiproject/sakai/commit/a3a54b1e408e73a06550e67e77552a1f55e1974f#diff-290359684d1bbbe62fe25dcaeb0f85fdL1302

Comment (ottenhoff) [reply edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java]: No, I don't see any need to use MathContext.DECIMAL128

**Comment by ottenhoff** [comment]
@ern there is no pattern of unit tests in edu-services/gradebook-service/hibernate/. Not easy for developers to add unit tests here without an existing pattern.

**Comment by juanjmerono** [comment]
I'm agree with @ottenhoff add a test here without a pattern it won't be easy. Please @jesusmmp review the codacy check before merge this. Thank you.

**Comment by ottenhoff** [comment]
I did quite a bit of testing locally and this works. I agree that we should add unit tests where possible but without a pattern, or someone to help create a new pattern, it's too much of an ask for this PR.

**Pull request merged**